File: app/views/comments.py
from app import app
from flask import session, redirect, request, jsonify
import html
from app.models.comments import add_comment, get_by_comment_id, dell_comment
from app.models.users import get_user_by_id


@app.route('/ajax_add_comment/', methods=['POST'])
def ajax_add_comment():
	if session.get('id_user_logged'):
		id_user = session.get('id_user_logged')
		id_post = request.form.get('id_post')
		text = html.escape(request.form.get('text').strip())

		if text:
			user = get_user_by_id(id_user)
			if user and user['active']:
				res = add_comment(id_user, id_post, text)
				if res:
					data = {
						'user_avatar': user['avatar'],
						'user_first_name': user['first_name'],
						'id_user': id_user,
						'id_post': id_post,
						'id_comment': res,
						'text': text
					}
					return jsonify(data)
	return "False"


@app.route('/ajax_dell_comment/', methods=['POST'])
def ajax_dell_comment():
	if session.get('id_user_logged'):
		id_user = session.get('id_user_logged')
		id_comment = request.form.get('id_comment')

		res = get_by_comment_id(id_comment)
		if res and res['id_user'] == id_user:
			dell = dell_comment(id_comment)
			if not dell:
				return "deleted"
	return "Fail"

File: app/static/js/comments.js
function add_comment(form) {
	let id_post = form.id_post.value;
	let text = form.comment.value;

	$.post('/ajax_add_comment/', {
		'id_post': id_post,
		'text': text
	}).done(function (resp) {
		form.comment.value = "";
		if (resp !== "False"){
			let all_comments = $('#all_comments' + resp.id_post);
			let data = "<div class=\"post-comment\" id='comment" + resp.id_comment + "'>" +
				"<img src=\"" + resp.user_avatar + "\" class=\"profile-photo-sm\" />" +
				"<a href=\"/user/id" + resp.id_user + "\" class=\"profile-link\">" + resp.user_first_name + "</a>" +
				"<p>&nbsp;&nbsp;" + resp.text + "</p>" +
				"<div class=\"btn text-red dell\" onclick=\"dell_comment(" + resp.id_comment + ")\">X</div>" +
			"</div>" +
			"<div class=\"line-divider\" id=\"divider" + resp.id_comment + "\"></div>";
			all_comments.prepend(data);
		}
	});
}

function dell_comment(id_comment) {
	$.post('/ajax_dell_comment/', {
		'id_comment': id_comment
	}).done(function (resp) {
		if (resp === "deleted") {
			let comment = $("#comment" + id_comment);
			let divider = $("#divider" + id_comment);

			comment.remove();
			divider.remove();
		}
	});
}
